src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -1724,11 +1723,12 @@ std::shared_ptr<IPayloadPool> DataReaderImpl::get_payload_pool()
     }
 
     PoolConfig config = PoolConfig::from_history_attributes(history_.m_att);
+    sample_pool_ = std::make_shared<detail::SampleLoanManager>(config, type_);
```

Comment:
Should we encapsulate this in an `if (!sample_pool_)`. My concern is to end up calling `get_payload_pool()` more than once, if not now in the future.